File: BlogPosts.Web/ClientApp/src/Components/Post.js
import React from 'react';
import { Link } from 'react-router-dom';
import format from 'date-fns/format';
const Post = ({ post }) => {
    const { text, dateSubmitted, title, comments, id } = post;
    return (

        <div className="card mb-4">
            <div className="card-body ">
                <h2 className='card-title' color='text-primary'>{title}</h2>
            </div>
            <p className='card-text'>{text.length < 200 ? text : text.substring(0, 200) + "..."}</p>
            <span>{comments.length} comment(s)</span>

            <Link to={`/ViewBlog/${id}`}>
                <button className='btn btn-primary '>ReadMore</button>
            </Link>

            <div className='card-footer text-muted'>
                Posted on {format(new Date(dateSubmitted), 'EEEE LLLL do, R')}
            </div>
        </div>

    )

}
export default Post;

File: BlogPosts.Data/Post.cs
using System;
using System.Collections.Generic;

namespace BlogPosts.Data
{
    public class Post
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string Name { get; set; }
        public string Text { get; set; }
        public DateTime DateSubmitted { get; set; }
        public List<Comment>Comments { get; set; }
    }
}
